vendor/k8s.io/kubernetes/pkg/controller/podautoscaler/legacy_horizontal_test.go

```diff
@@ -472,7 +472,10 @@ func (tc *legacyTestCase) runTest(t *testing.T) {
 		if tc.finished {
 			return true, &v1.Event{}, nil
 		}
-		obj := action.(core.CreateAction).GetObject().(*v1.Event)
+		obj, ok := action.(core.CreateAction).GetObject().(*v1.Event)
```

Comment:
So the client-go code says the same event (without defining what `the same` means) is patched when it's has been already recorded [1]. The fix would be then to allow to accept even Patch events. Hoping the Reason will stay the same. Or, just ignore any patch event.

[1] https://github.com/kubernetes/client-go/blob/master/tools/record/event.go#L214-L216